tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Properties/InterceptingProjectProperties/BuildPropertyPage/NameQuotedValuePairListEncodingTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.InterceptingProjectProperties.BuildPropertyPage
+{
+    public class NameQuotedValuePairListEncodingTests
+    {
+        [Theory]
+        [InlineData("key1=value1", "key1=\"value1\"")]
+        [InlineData("key1=value1,key2=value2", "key1=\"value1\",key2=\"value2\"")]
+        [InlineData("a=b,c=easy,as=123", "a=\"b\",c=\"easy\",as=\"123\"")]
+        [InlineData("key1=value1,key2=value2,key3=value3", "key1=\"value1\",key2=\"value2\",key3=\"value3\"")]
+        [InlineData("something=equals=this", "something=\"equals/=this\"")]
+        [InlineData("oh=hey=there,hi=didnt=see,you=there", "oh=\"hey/=there\",hi=\"didnt/=see\",you=\"there\"")]
+        [InlineData("path=/path/to/somewhere/", "path=\"//path//to//somewhere//\"")]
+        [InlineData("file=/path/is/here/", "file=\"//path//is//here//\"")]
+        [InlineData("files=path=/path/to/somewhere/", "files=\"path/=//path//to//somewhere//\"")]
+        [InlineData("a=1,a=2", "a=\"1\",a=\"2\"")]
+        [InlineData("key1  =value1  ", "key1  =\"value1  \"")]
+        [InlineData("  key1=  value1", "  key1=\"  value1\"")]
+        [InlineData(" key1 = value1 ", " key1 =\" value1 \"")]
+        [InlineData(" = ", " =\" \"")]
+        [InlineData("key1=a b c d", "key1=\"a b c d\"")]
+        [InlineData("key1=a=b=c=d", "key1=\"a/=b/=c/=d\"")]
+        [InlineData("key1=a=b\"\"c=d", "key1=\"a/=b/\"/\"c/=d\"")]
+        [InlineData("key1=\"ab\"\"cd", "key1=\"/\"ab/\"/\"cd\"")]
+        [InlineData("key1=\"\"ab\"\"cd\"\"ef", "key1=\"/\"/\"ab/\"/\"cd/\"/\"ef\"")]
+        [InlineData("key1=\"\"ab\"\"cd", "key1=\"/\"/\"ab/\"/\"cd\"")]
+        public void ValidNameQuotedValuePairListEncoding(string input, string expectedOutput)
+        {
+            NameQuotedValuePairListEncoding _encoding = new();
+            Assert.Equal(expected: expectedOutput, actual: _encoding.Format(_encoding.Parse(input)));
```

Comment:
I feel like this test is too coarse and could miss bugs. It's testing both the parsing and reformatting at once, without really inspecting the intermediate value. It's the intermediate value that the UI will use, so it's quite important that it's correct.

By testing two things at once here it's possible that a bug in the first is undone in the second, and therefore goes unnoticed.

Something like this (written in a browser, so may not compile, but should get the idea across):

```c#
[Theory]
[InlineData("a=\"1\"", new[] { "a", "1" })]
[InlineData("a=\"1\",b=\"2\"", new[] { "a", "1", "b", "2" })]
public void ValidNameQuotedValuePairListEncoding(string encoded, string[] pairs)
{
    NameQuotedValuePairListEncoding _encoding = new();

    // Try parsing
    var actualPairs = _encoding.Parse(encoded);

    Assert.Equal(pairs, actualPairs.SelectMany(pair => new[] { pair.Name, pair.Value })); 

    // Try formatting
    var actualEncoded = _encoding.Format(ToNameValues(pairs));

    Assert.Equal(encoded, actualEncoded);

    static IEnumerable<(string Name, string Value)> ToNameValues(IEnumerable<string> pairs)
    {
        using var e = pairs.GetEnumerator();
        while (e.MoveNext())
        {
            var name = e.Current;
            Assert.True(e.MoveNext());
            var value = e.Current;
            yield return (name, value);
        }
    }
}
```